### PR TITLE
Fix asyncio deprecation asyncio.Task.all_tasks

### LINKED ISSUE #3297: Cannot shutdown on python 3.9 due to asyncio deprecation
### Bug description
When using jupyterhub with python3.9, you will not be able to shutdown (e.g. via ctrl+c). When attempting to stop asyncio tasks, you will get `type object '_asyncio.Task' has no attribute 'all_tasks'` error. `asyncio.Task.all_tasks()` and `current_task()` have been deprecated since python 3.7 and is fully removed in 3.9.  So all references to `asyncio.Task.all_tasks()` should change to `asyncio.all_tasks()`.

I have a fix that will accommodate python3.6 and later. It's quite a simple fix. I hope it fits the conventions that this project uses. btw, if support for python3.6 can be dropped, then the change is even simpler and without any clutter.

#### Expected behaviour
jupyterhub should terminate gracefully like:
```
^C[C 2020-12-02 17:50:48.625 JupyterHub app:2826] Received signal SIGINT, initiating shutdown...
[I 2020-12-02 17:50:48.626 JupyterHub app:2467] Cleaning up single-user servers...
[I 2020-12-02 17:50:48.627 JupyterHub proxy:735] Cleaning up proxy[9371]...
17:50:48.633 [ConfigProxy] warn: Terminated
[I 2020-12-02 17:50:48.638 JupyterHub app:2499] ...done
```

#### Actual behaviour
```
^C[C 2020-12-02 16:16:45.801 JupyterHub app:2810] Received signal SIGINT, initiating shutdown...
ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-8' coro=<JupyterHub.shutdown_cancel_tasks() done, defined at /opt/miniconda3/envs/jupyterhub/lib/python3.9/site-packages/jupyterhub/app.py:2808> exception=AttributeError("type object '_asyncio.Task' has no attribute 'all_tasks'")>
Traceback (most recent call last):
  File "/opt/miniconda3/envs/jupyterhub/lib/python3.9/site-packages/jupyterhub/app.py", line 2812, in shutdown_cancel_tasks
    t for t in asyncio.Task.all_tasks() if t is not asyncio.Task.current_task()
AttributeError: type object '_asyncio.Task' has no attribute 'all_tasks'
```

### How to reproduce
I'm using conda to install python and jupyterhub
```
conda create -n jupyterhub python=3.9 jupyterhub
jupyterhub
ctrl-c
```

### Your personal set up
 - OS:
Ubuntu  20.04.1
 - Version:
jupyterhub 1.2.2
 - Configuration:
using conda, as shown above.

### PR DESCRIPTION
**Pull request by coffeebenzene**
This is to fix #3297

### CODE REVIEW

**Review by consideRatio — APPROVED** [review]
Beautiful! This LGTM! Thanks for a well written bug report as and adding the comment which help us remove this logic when we no longer support Py36 as well! :heart:

File: jupyterhub/app.py [hunk 1]
```diff
@@ -29,6 +29,13 @@
 if sys.version_info[:2] < (3, 3):
     raise ValueError("Python < 3.3 not supported: %s" % sys.version)
 
+# For compatibility with python versions 3.6 or earlier.
+# asyncio.Task.all_tasks() is fully moved to asyncio.all_tasks() starting with 3.9. Also applies to current_task.
+try:
+    asyncio.all_tasks
+except AttributeError as e:
+    asyncio.all_tasks = asyncio.Task.all_tasks
```
Comment (manics) [inline jupyterhub/app.py:37]: This is monkeypatching the asyncio Python module in place. It's probably fine, though monkeypatching can occasionally lead to some weird corner cases or confuse people debugging problems. What do you think about assigning it to a variable instead, e.g.
```py
try:
    asyncio_all_tasks = asyncio.all_tasks
except AttributeError as e:
    asyncio_all_tasks = asyncio.Task.all_tasks
    ...
```

Comment (consideRatio) [reply jupyterhub/app.py:37]: I like the PR's compromise over this suggested compromise unless we actually override something in asyncio but only adds an entry, which I assume we don't.

I mostly consider this from a readability standpoint, and have little experience with issues caused by modifying classes.

Comment (minrk) [reply jupyterhub/app.py:37]: Even though I think it's highly unlikely to cause an issue, I think we should definitely go with @manics suggestion. Assigning attributes on a stdlib module isn't something i think we should be doing.

Comment (coffeebenzene) [reply jupyterhub/app.py]: Ah ok, I've changed the monkeypatching to using variables instead, so we don't modify anything on modules.

**Review by coffeebenzene — COMMENTED** [review]
Oof I realise that my comment replies were pending instead of being submitted, so they were hidden

**Comment by minrk** [comment]
Perfect thank you!

**Pull request merged**